src/Symfony/Component/Messenger/Worker.php

```diff
@@ -282,4 +326,79 @@ public function getMetadata(): WorkerMetadata
     {
         return $this->metadata;
     }
+
+    public function handleFutures(string $transportName, $parallelProcessLimit): void
+    {
+        $toHandle = self::$futures;
+        self::$futures = [];
+
+        if (!$toHandle) {
+            return;
+        }
+
+        $futuresReceived = [];
+        $envelopesAssociated = [];
+
+        foreach ($toHandle as $combo) {
+            $futuresReceived[] = $combo[0];
+            $envelopesAssociated[] = $combo[1];
+        }
+
+        [$errorsFromAwait, $executions] = awaitAll($futuresReceived);
```

Comment:
If I understand correctly. This will prevent any new jobs from being processed until the current jobs are finished processing.

Further, if another fiber comes along for this same worker, it will see these same futures in self::futures and wait for them as well.

What you probably want to do, is iterate over any completed futures, and only process those that have already completed. Otherwise, you won't be able to continue past this line until all work is complete.